Microsoft.Toolkit/Extensions/ISettingsStorageHelperExtensions.cs

```diff
@@ -0,0 +1,81 @@
+        /// If the key is not found, the fallback value will be used instead.
+        /// </summary>
+        /// <typeparam name="TKey">The type of key used to lookup the object.</typeparam>
+        /// <typeparam name="TValue">The type of object value expected.</typeparam>
+        /// <param name="storageHelper">The storage helper instance fo read from.</param>
+        /// <param name="key">The key of the target object.</param>
+        /// <param name="value">The value of the target object, or the fallback value.</param>
+        /// <param name="fallback">An alternative value returned if the read fails.</param>
+        /// <returns>A boolean indicator of success.</returns>
+        public static bool TryGetValueOrDefault<TKey, TValue>(this ISettingsStorageHelper<TKey> storageHelper, TKey key, out TValue? value, TValue? fallback = default)
+            where TKey : notnull
+        {
+            if (storageHelper.TryRead<TValue>(key, out TValue? storedValue))
+            {
+                value = storedValue;
+                return true;
+            }
+            else
+            {
+                value = fallback;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Read the key in the storage helper instance and get the value.
+        /// </summary>
+        /// <typeparam name="TKey">The type of key used to lookup the object.</typeparam>
+        /// <typeparam name="TValue">The type of object value expected.</typeparam>
+        /// <param name="storageHelper">The storage helper instance fo read from.</param>
+        /// <param name="key">The key of the target object.</param>
+        /// <returns>The value of the target object</returns>
+        /// <exception cref="KeyNotFoundException">Throws when the key is not found in storage.</exception>
+        public static TValue? Read<TKey, TValue>(this ISettingsStorageHelper<TKey> storageHelper, TKey key)
+            where TKey : notnull
+        {
+            if (storageHelper.TryRead<TValue>(key, out TValue? value))
+            {
+                return value;
+            }
+            else
+            {
+                throw new KeyNotFoundException();
+            }
+        }
+
+        /// <summary>
+        /// Deletes a key from storage.
+        /// </summary>
+        /// <typeparam name="TKey">The type of key used to lookup the object.</typeparam>
+        /// <param name="storageHelper">The storage helper instance to delete from.</param>
+        /// <param name="key">The key of the target object.</param>
+        /// <exception cref="KeyNotFoundException">Throws when the key is not found in storage.</exception>
+        public static void Delete<TKey>(this ISettingsStorageHelper<TKey> storageHelper, TKey key)
+            where TKey : notnull
+        {
+            if (!storageHelper.TryDelete(key))
+            {
+                throw new KeyNotFoundException();
```

Comment:
We could make this easier to debug for consumers and also with better codegen with this helper:

```csharp
private static void ThrowKeyNotFoundException<TKey>(TKey key)
{
    throw new KeyNotFoundException($"The given key '{key}' was not present");
}
```

And then replace these throws with just `ThrowKeyNotFoundException(key)`.
In the previous method, which has to return a value, you can just add `return default` right after that call.

For more info as to why the codegen is better, if you're interested, [I've written some docs here](https://docs.microsoft.com/en-us/windows/communitytoolkit/developer-tools/throwhelper).